android/app/build.gradle

```
@@ -80,11 +80,17 @@ android {
         multiDexEnabled true
 
         if (System.getenv()["KETTLE_API_KEY"]) {
-            manifestPlaceholders = [KETTLE_API_KEY: System.getenv()["KETTLE_API_KEY"] as String]
+            manifestPlaceholders = [
+                KETTLE_API_KEY: System.getenv()["KETTLE_API_KEY"] as String,
+                ENABLE_ACTIVITY_RECOGNITION: "true"
+            ]
         } else {
-            // If this value is not set, an exception will be raised at runtime. Refer to the documentation at https://developer.kogenta.com/docs/react-native/add-kettle#android for more information.
+            // If `KETTLE_API_KEY` value is not set, an exception will be raised at runtime. Refer to the documentation at https://developer.kogenta.com/docs/react-native/add-kettle#android for more information.
             // NOTE: To avoid breaking the app and to prevent the SDK from initializing, you can use a value other than an API key.
-            manifestPlaceholders = [KETTLE_API_KEY: "NoKettleIdNeeded"]
+            manifestPlaceholders = [
+                KETTLE_API_KEY: "NoKettleIdNeeded",
+                ENABLE_ACTIVITY_RECOGNITION: "false"
```

```suggestion
                ACTIVITY_RECOGNITION_NODE_VALUE: activityRecognitionBool ? "merge" : "remove"
```

And also add this further up:
```
def activityRecognitionEnv = System.getenv("ENABLE_ACTIVITY_RECOGNITION")
def activityRecognitionBool = activityRecognitionEnv != null && activityRecognitionEnv.equalsIgnoreCase("true")
```

And then set `ENABLE_ACTIVITY_RECOGNITION` in the env files

---

Is true is by using tool:node, I made the fix. We do not need an env var for this, we just use if kettle is set or not.

---

It is removed when the Kettle SDK Key is not present, which is the desired. is Inside a condition. Not sure if I understood you, if is something else.

---

Sorry yes it should be working now. Somehow I didn't notice the System.getenv()["KETTLE_API_KEY"] check earlier..